run-dljc.sh

```diff
@@ -100,179 +116,107 @@ fi
-  else
-      DLJC_CMD="${DLJC} -t wpi --cleanCmd \"${CLEAN_CMD}\""
-      if [ ! "x${CHECKERS}" = "x" ]; then
-	  TMP="${DLJC_CMD} --checker ${CHECKERS}"
-          DLJC_CMD="${TMP}"
-      fi
-      if [ ! "x${CHECKER_LIB}" = "x" ]; then
-	  TMP="${DLJC_CMD} --lib ${CHECKER_LIB}"
-	  DLJC_CMD="${TMP}"
-      fi
-      
-      if [ ! "x${STUBS}" = "x" ]; then
-	  TMP="${DLJC_CMD} --stubs ${STUBS}"
-	  DLJC_CMD="${TMP}"
-      fi
-      
-      if [ ! "x${QUALS}" = "x" ]; then
-	  TMP="${DLJC_CMD} --quals ${QUALS}"
-	  DLJC_CMD="${TMP}"
-      fi
-      
-      if [ ! "x${TOUT}" = "x" ]; then
-          TMP="${DLJC_CMD}"
-          DLJC_CMD="timeout ${TOUT} ${TMP}"
-          echo "setting timeout to ${TOUT}"
-      fi
-      
-      TMP="${DLJC_CMD} -- ${BUILD_CMD}"
-      DLJC_CMD="${TMP}"
-      
-      # ensure the project is clean before invoking DLJC
-      eval ${CLEAN_CMD} < /dev/null
-
-      echo ${DLJC_CMD}
-      
-      eval ${DLJC_CMD} < /dev/null
-      
-      if [[ $? -eq 124 ]]; then
-          echo "dljc timed out for ${REPO_NAME}"
-          echo "dljc timed out for ${REPO_NAME}" > ../../../${OUTDIR}-results/${REPO_NAME}-wpi.log
-	  USABLE="no"
-      else 
-          if [ -f dljc-out/wpi.log ]; then
-              cp dljc-out/wpi.log ../../../${OUTDIR}-results/${REPO_NAME}-wpi.log
-              echo "${REPO},${HASH}" >> ../../../${OUTDIR}-results/interesting-results.csv
-	      USABLE="yes"
-          else
-              # if this last run was under Java 11, try to run
-              # under Java 8 instead
-              if [ "${JAVA_HOME}" = "${JAVA11_HOME}" ]; then
-                  export JAVA_HOME="${JAVA8_HOME}"
-                  echo "couldn't build using Java 11; trying Java 8"
-                  configure_and_exec_dljc
-              else
-                  echo "dljc could not run the build successfully" > ../../../${OUTDIR}-results/${REPO_NAME}-wpi.log
-		  USABLE="no"
-              fi
-          fi
-      fi
-  fi
-  export JAVA_HOME="${JAVA11_HOME}"
-}
+export PATH="${JAVA_HOME}/bin:${PATH}"
+
+mkdir -p "${OUTDIR}"
+mkdir -p "${OUTDIR}-results"
+
+pushd "${OUTDIR}" || exit 5
 
 while IFS='' read -r line || [ "$line" ]
 do    
     REPOHASH=${line}
 
-    REPO=`echo ${REPOHASH} | awk '{print $1}'`
-    HASH=`echo ${REPOHASH} | awk '{print $2}'`
+    REPO=$(echo "${REPOHASH}" | awk '{print $1}')
+    HASH=$(echo "${REPOHASH}" | awk '{print $2}')
 
-    REPO_NAME=`echo ${REPO} | cut -d / -f 5`
+    REPO_NAME=$(echo "${REPO}" | cut -d / -f 5)
+    REPO_NAME_HASH="${REPO_NAME}-${HASH}"
 
-    # need a layer in the file structure that prevents
-    # two repos with the same name from colliding
-    if [ ! -d ${REPO_NAME}-${HASH} ]; then
-        mkdir ${REPO_NAME}-${HASH}
-    fi
+    # Use repo name and hash, but not owner.  We want
+    # repos that are different but have the same name to be treated
+    # as different repos, but forks with the same content to be skipped.
```

Comment:
How are they skipped?  I don't see the logic later on that does this, but maybe it is implicit.

Reply:
It's implicit, and they're not really *skipped*, but rather clobber each other. That is, every one of such forks will be typechecked, but results from later ones will overwrite results from earlier ones (so the early ones, from the output's perspective, have been "skipped"). This behavior is kind of bad (repeating useless work). I'll add a test so that the useless work is skipped and the first set of results is used instead.

Reply:
Actually, I've changed my mind about this. Skipping repositories with identical name-hash combinations isn't possible without a different strategy for re-running experiments using the same set of downloaded repos, which is a behavior I want to preserve. The current behavior is a bit wasteful but does the correct thing. If you think it's very important to change, we can discuss it more.